packages/flutter_availability/lib/src/ui/widgets/template_week_overview.dart

```diff
@@ -0,0 +1,178 @@
+
+    var templateData = template.data;
+
+    return Column(
+      crossAxisAlignment: CrossAxisAlignment.start,
+      children: [
+        Text(
+          translations.weekTemplateOverviewTitle,
+          style: textTheme.titleMedium,
+        ),
+        const SizedBox(height: 8),
+        DecoratedBox(
+          decoration: BoxDecoration(
+            color: colors.templateWeekOverviewBackgroundColor ??
+                theme.colorScheme.primaryContainer,
+            borderRadius: BorderRadius.circular(10),
+            border: Border.all(
+              color: theme.dividerColor,
+            ),
+          ),
+          child: Column(
+            children: [
+              for (var day in WeekDay.values) ...[
+                _TemplateDayDetailRow(
+                  dayName: dayNames[day.index],
+                  dayData:
+                      templateData.containsKey(day) ? templateData[day] : null,
+                  isOdd: day.index.isOdd,
+                ),
+              ],
+            ],
+          ),
+        ),
+      ],
+    );
+  }
+}
+
+class _TemplateDayDetailRow extends StatelessWidget {
+  const _TemplateDayDetailRow({
+    required this.dayName,
+    required this.dayData,
+    required this.isOdd,
+  });
+
+  /// The name of the day
+  final String dayName;
+
+  /// There odd rows do not have a background color
+  /// This causes a layered effect
+  final bool isOdd;
+
+  /// The data of the day
+  final DayTemplateDataViewModel? dayData;
+
+  @override
+  Widget build(BuildContext context) {
+    var theme = Theme.of(context);
+    var textTheme = theme.textTheme;
+    var availabilityScope = AvailabilityScope.of(context);
+    var options = availabilityScope.options;
+    var translations = options.translations;
+
+    var startTime = dayData?.startTime;
+    var endTime = dayData?.endTime;
+    var dayPeriod = startTime != null && endTime != null
+        ? "${translations.timeFormatter(context, startTime)} - "
+            "${translations.timeFormatter(context, endTime)}"
+        : translations.unavailable;
+
+    return Container(
+      decoration: BoxDecoration(
+        color: isOdd ? Colors.white : null,
+        border: isOdd
+            ? Border(
+                left: BorderSide(
+                  color: theme.dividerColor,
+                ),
+                right: BorderSide(
+                  color: theme.dividerColor,
+                ),
+              )
+            : null,
+      ),
+      padding: const EdgeInsets.all(16),
+      child: Column(
+        children: [
+          Row(
+            mainAxisAlignment: MainAxisAlignment.spaceBetween,
+            children: [
+              Text(dayName, style: textTheme.bodyLarge),
+              Text(dayPeriod, style: textTheme.bodyLarge),
+            ],
+          ),
+          // for each break add a line
+          for (var dayBreak
+              in dayData?.breaks ?? <AvailabilityBreakModel>[]) ...[
```

Comment:
Retrieve breaks as a list before entering the layout. Try to separate logic as much as possible